src/java/org/apache/cassandra/db/transform/BaseIterator.java

```diff
@@ -83,14 +87,31 @@
 
     public final void close()
     {
+        // If close has already been called we want to ignore other calls
+        if (closed)
```

Comment:
Unless I am missing sthg: I'd wrap everything in a try-finally block and toggle `closed` to true only there. That is when it is indeed closed and not 'closing' as it is at this point. I can see the code _tries_ to catch/handle all failures but `runOnClose()` is abstract to be implemented in any way.

That would prevent a failed close operation to toggle the `closed` variable erroneously.

Reply:
If the iterator has been registered in multiples `try-with-resources`  (the root of the problem we are trying to solve) and a bad implementation of `runOnClose` fire an Exception, the next `try-with-resources` will hit the same problem and will hide the original one. No matter how the execution of `close` end up, I do not think that there are any scenarios in which you want to retry closing the iterator. Which is why the patch toggle `closed` as early as possible.